src/transform/src/analysis.rs

```diff
@@ -396,16 +395,17 @@ pub mod common {
             // Attempt optimistic analysis, and if that fails go pessimistic.
             let update = A::lattice()
                 .filter(|_| self.allow_optimistic)
-                .and_then(|lattice| {
+                .map(|lattice| {
                     for _ in exprs.iter() {
                         self.results.push(lattice.top());
                     }
                     self.analyse_optimistic(exprs, 0, exprs.len(), depends, &*lattice)
-                        .ok()
+                        .is_ok()
```

Comment:
Maybe a nit, but this does lock in "bool" as the only return type. Certainly true at the moment, no plans to change, but .. if we do we'll have to walk this back. Is there motivation for the change other than to spell out right here that the specific return type of `analyse_optimistic` is `Result<bool, _>` rather than some other type?

Reply:
The return type of `analyse_optimistic` is not `Result<bool, _>` anymore in this PR, but `Result<(), ()>`. The motivation was that both `analize_optimistic` and `analyze_pessimistic` only ever returned true when successful and their documentation did not mention what that bool signifies. Is there some meaning to the value of that boolean that happens to be "always yes" now?

Reply:
> Maybe a nit, but this does lock in "bool" as the only return type. 

If I understand correctly the opposite is true. The code on main requires that `.ok()` returns an `Option<bool>`, which in turn locks in `bool` as the only return type.

The code in this PR uses `.is_ok()` which means that `analyze_optimistic` can return something more complicated in its `Ok(_)` variant but this function doesn't care. Is this what you mean by locking it in?

Reply:
Ah, looking closer I understand you mean now. The `bool` returned  by `analyze_optimistic` and `analyze_pessimistic` has the meaning defined by the documentation of `AnalysisBundle::analyse` method, which says "Result indicates whether new information was produced for `exprs.last()`.".

That makes sense, and under that light I agree that the bool return value should stay! I will revert that part of the code and add a doc comment.

Reply:
Reverted, thanks for taking a look!